overlord/configstate/configstate.go

```diff
@@ -63,24 +85,20 @@ func Configure(st *state.State, snapName string, patch map[string]interface{}, f
 	} else if len(patch) > 0 {
 		contextData = map[string]interface{}{"patch": patch}
 	}
-	var summary string
+
 	if hooksup.Optional {
 		summary = fmt.Sprintf(i18n.G("Run configure hook of %q snap if present"), snapName)
-	} else {
-		summary = fmt.Sprintf(i18n.G("Run configure hook of %q snap"), snapName)
-	}
-
-	// configuration of "core" is handled differently
-	if snapName == "core" {
-		// TODO: respect IgnoreHookError and TrackHookError ?
-		t := st.NewTask("configure-snapd", summary)
-		t.Set("patch", patch)
-		if v, ok := contextData["use-defaults"].(bool); v && ok {
-			t.Set("use-defaults", true)
-		}
-		return state.NewTaskSet(t)
 	}
 
 	task := hookstate.HookTask(st, summary, hooksup, contextData)
 	return state.NewTaskSet(task)
 }
+
+// Configure returns a taskset to apply the given configuration patch.
+func Configure(st *state.State, snapName string, patch map[string]interface{}, flags int) *state.TaskSet {
+	// configuration for "core" is handled internally
+	if snapName == "core" {
+		return configureSnapd(st, snapName, patch, flags)
```

Comment:
Total nitpick: we know what we want to configure here, so passing snapName feels a bit redundant if the method is specialized for snapd.